pkg/controller/dataimportcron-controller.go

```diff
@@ -702,10 +711,19 @@ func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Contro
 	return nil
 }
 
-func (r *DataImportCronReconciler) cronJobExists(ctx context.Context, cron *cdiv1.DataImportCron) bool {
+func (r *DataImportCronReconciler) cronJobExistsAndUpdated(ctx context.Context, cron *cdiv1.DataImportCron) (bool, error) {
 	var cronJob batchv1.CronJob
 	cronJobNamespacedName := types.NamespacedName{Namespace: r.cdiNamespace, Name: GetCronJobName(cron)}
-	return r.uncachedClient.Get(ctx, cronJobNamespacedName, &cronJob) == nil
+	err := r.uncachedClient.Get(ctx, cronJobNamespacedName, &cronJob)
+	podSpec := &cronJob.Spec.JobTemplate.Spec.Template.Spec
+	if err == nil && len(podSpec.Containers) > 0 && podSpec.Containers[0].Image != r.image {
+		podSpec.Containers[0].Image = r.image
```

Comment:
Maybe Update() over newCronJob().Spec != foundCron.Spec?
It is possible at some point in the future we decided to change some fields in the cronjob object too, and want to sync existing cronjobs to those values

Reply:
@akalenyu tried this change and it seems that when CronJob is created based on newCronJob(), k8s patches several fields of the spec (dnsPolicy, schedulerName, serviceAccount etc.) so we cannot simply compare (DeepEqual) the current and the desired spec.

Reply:
Use your tip, thanks @akalenyu !